packages/number-input/src/Input/Input.styles.ts

```diff
@@ -0,0 +1,258 @@
+import { css } from '@leafygreen-ui/emotion';
+import { createUniqueClassName, Theme } from '@leafygreen-ui/lib';
+import { palette } from '@leafygreen-ui/palette';
+import {
+  focusRing,
+  fontFamilies,
+  hoverRing,
+  spacing,
+  transitionDuration,
+  typeScales,
+} from '@leafygreen-ui/tokens';
+
+import { Size, State } from '../NumberInput/NumberInput.types';
+
+/**
+ * Adds an inset box shadow to hide the UA background styles for autofilled inputs
+ */
+const autofillShadowOverride = (color: string) => `0 0 0 100px ${color} inset`;
+
+export const wrapperClassName = createUniqueClassName('number-input-wrapper');
+
+export const inputBaseStyles = css`
+  all: unset;
+  font-family: ${fontFamilies.default};
+  font-weight: normal;
+  width: 100%;
+  font-size: ${typeScales.body1.fontSize}px;
+  line-height: ${typeScales.body1.lineHeight}px;
+  height: inherit;
+  box-sizing: border-box;
+  transition: ${transitionDuration.default}ms ease-in-out;
+  transition-property: background-color;
+
+  &::-webkit-outer-spin-button,
+  &::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  &[type='number'] {
+    -moz-appearance: textfield;
+  }
+
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+
+    &:hover,
+    &:active {
+      box-shadow: none;
+    }
+  }
+`;
+
+export const inputErrorPaddingTransitionStyles = css`
+  transition-property: padding;
+`;
+
+export const inputAnimateStyles = css`
+  .${wrapperClassName}:hover &,
+  .${wrapperClassName}:focus-within & {
+    padding-right: ${spacing[1] * 6}px;
+  }
+`;
+
+export const inputThemeStyles: Record<Theme, string> = {
+  [Theme.Light]: css`
+    &::placeholder {
+      color: ${palette.gray.light1};
+    }
+
+    &:-webkit-autofill {
+      -webkit-text-fill-color: ${palette.black};
+      border: 0;
+      box-shadow: ${autofillShadowOverride(palette.white)};
+
+      &:focus-visible,
+      &:hover {
+        -webkit-text-fill-color: ${palette.black};
+        box-shadow: ${autofillShadowOverride(palette.white)};
+      }
+    }
+  `,
+  [Theme.Dark]: css`
+    &::placeholder {
+      color: ${palette.gray.dark1};
+    }
+
+    &:-webkit-autofill {
+      -webkit-text-fill-color: ${palette.gray.light2};
+      border: 0;
+      box-shadow: ${autofillShadowOverride(palette.gray.dark4)};
+
+      &:focus-visible,
+      &:hover {
+        -webkit-text-fill-color: ${palette.gray.light2};
+        box-shadow: ${autofillShadowOverride(palette.gray.dark4)};
+      }
+    }
+  `,
+};
+
+export const sizeInputStyles: Record<Size, string> = {
+  [Size.XSmall]: css`
+    padding: 0 0 0 ${spacing[1] * 3}px;
+  `,
+  [Size.Small]: css`
+    padding: 0 0 0 ${spacing[1] * 3}px;
+  `,
+  [Size.Default]: css`
+    padding: 0 0 0 ${spacing[1] * 3}px;
+  `,
+};
```

Comment:
Should these values be different based on size?

Reply:
Otherwise, they could be all in a `baseStyles`

Reply:
removing this